front/src/components/Messages.js

```diff
@@ -55,14 +55,21 @@ const MessagesDiv = styled.div`
 
 
 const Messages = (props) => {
+  const [isNavigationMode, setNavigationMode] = useState()
   const messagesEndRef = useRef(null)
 
   useEffect(() => {
-    messagesEndRef.current.scrollIntoView({ behavior: "smooth" })
+    !isNavigationMode && messagesEndRef.current.scrollIntoView({ behavior: "smooth" })
   }, [props.messages])
 
+  const handleScroll = (scrollEvent) => {
+      var element = scrollEvent.currentTarget
+      var isBottom = element.scrollTop + element.offsetHeight === element.scrollHeight
```

Comment:
avoid `var`. Use `const` for variables that you don't mutate and `let` for those that you do.